components/client/src/main/java/com/hotels/styx/client/OriginsInventory.java

```diff
@@ -476,54 +466,33 @@ private OriginState state() {
         }
     }
 
-    public static Builder newOriginsInventoryBuilder(BackendService backendService) {
-        return new Builder(backendService);
+    public static Builder newOriginsInventoryBuilder(Id appId) {
+        return new Builder(appId);
     }
 
     /**
      * A builder for {@link com.hotels.styx.client.OriginsInventory}.
      */
     public static class Builder {
-        private final BackendService backendService;
-        private final HttpConfig.Builder httpConfigBuilder = newHttpConfigBuilder();
-        private OriginHealthStatusMonitor healthStatusMonitor;
-        private OriginHealthStatusMonitor.Factory originHealthStatusMonitorFactory;
+        private final Id appId;
+        private OriginHealthStatusMonitor originHealthMonitor = new NoOriginHealthStatusMonitor();
         private MetricRegistry metricsRegistry = new CodaHaleMetricRegistry();
-        private String version = "";
-        private ConnectionPool.Factory connectionPoolFactory;
-        private Connection.Factory connectionFactory;
         private EventBus eventBus = new EventBus();
-        private int clientWorkerThreadsCount = 1;
-        private OriginStatsFactory originStatsFactory;
-
-
-        public Builder healthStatusMonitor(OriginHealthStatusMonitor healthStatusMonitor) {
-            this.healthStatusMonitor = healthStatusMonitor;
-            return this;
-        }
-
-        public Builder originHealthStatusMonitorFactory(OriginHealthStatusMonitor.Factory originHealthStatusMonitorFactory) {
-            this.originHealthStatusMonitorFactory = originHealthStatusMonitorFactory;
-            return this;
-        }
+        private ConnectionPool.Factory connectionPoolFactory;
+        private Set<Origin> initialOrigins = emptySet();
 
         public Builder metricsRegistry(MetricRegistry metricsRegistry) {
             this.metricsRegistry = metricsRegistry;
             return this;
         }
 
-        public Builder version(String version) {
-            this.version = version;
-            return this;
-        }
-
         public Builder connectionPoolFactory(ConnectionPool.Factory connectionPoolFactory) {
             this.connectionPoolFactory = connectionPoolFactory;
             return this;
         }
 
-        public Builder connectionFactory(Connection.Factory connectionFactory) {
-            this.connectionFactory = connectionFactory;
+        public Builder originHealthMonitor(OriginHealthStatusMonitor originHealthMonitor) {
+            this.originHealthMonitor = originHealthMonitor;
```

Comment:
We should check this for null here, to avoid throwing an exception in the `build` method.

Reply:
ok